base/sparse/sparsematrix.jl

```diff
@@ -134,30 +134,54 @@ function Base.show(io::IOContext, S::SparseMatrixCSC)
     end
 
     limit::Bool = get(io, :limit, false)
-    if limit
-        rows = displaysize(io)[1]
-        half_screen_rows = div(rows - 8, 2)
+    rows = displaysize(io)[1] - 4 # -4 from [Prompt, header, newline after elements, new prompt]
+    rows <= 2 && return
```

Comment:
I don't think you're right about that. The existing code prints something `if k < half_screen_rows || k > nnz(S)-half_screen_rows` or `k == half_screen_rows`. Are there any situations in which none of those conditions ever happens?